### DISCUSSION TITLE
Dashboard Export for sharing externally using the API

### DISCUSSION THREAD

**Discussion by muralikanagala** [grafana/grafana#45167] [Dashboards, Panels & Visualizations]
Hi All - I am working on an automation to export dashboards from one Grafana instance to another. I am using the Grafana REST API to export the dashboard and import to another Grafana instance. It was working earlier as the Panel sections of the JSONs had the data source name which I maintain the same between the Grafana instances. But with recent changes in the JSON export I see only the data source type and the uid which are not consistent in my use case. 

But if I export from the UI enabling "Export for sharing externally" it gets rid of the IDs and uses the data source names. 

Is there any API path/parameter to get the JSON exported for sharing externally?

**Comment by pruthvi247** [comment]
@muralikanagala  I have come across similar use case, any updates on this issue ?? or any work around you could suggest. Thanks !

**Comment by suikast42** [comment]
I run in the same Issue. 

I write a rest tool which exports the dashbaords and do some replacements during the export.

I replace the of datasource with Datasource-[datasource-name].

The importer queries at fist the datasources on the "destination grafana" and then replace the Datasource-[datasource-name] with the uid. That precedes sme datasource names in both grafana deployments.

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
